secrets/secrets.go

```diff
@@ -71,30 +74,91 @@ var NewKeeper = newKeeper
 // newKeeper creates a Keeper.
 func newKeeper(k driver.Keeper) *Keeper {
 	return &Keeper{
-		k: k,
-		tracer: &oc.Tracer{
-			Package:        pkgName,
-			Provider:       oc.ProviderName(k),
-			LatencyMeasure: latencyMeasure,
-		},
+		k:      k,
+		tracer: otel.GetTracerProvider().Tracer(pkgName),
 	}
 }
 
 const pkgName = "gocloud.dev/secrets"
 
 var (
-	latencyMeasure = oc.LatencyMeasure(pkgName)
+	// Meter is the OpenTelemetry meter for this package
+	meter = otel.GetMeterProvider().Meter(pkgName)
 
-	// OpenCensusViews are predefined views for OpenCensus metrics.
-	// The views include counts and latency distributions for API method calls.
-	// See the example at https://godoc.org/go.opencensus.io/stats/view for usage.
-	OpenCensusViews = oc.Views(pkgName, latencyMeasure)
+	// latencyHistogram measures the latency of method calls
+	latencyHistogram metric.Float64Histogram
+
+	// completedCallsCounter counts the number of method calls
+	completedCallsCounter metric.Int64Counter
 )
 
+func init() {
+	var err error
+	latencyHistogram, err = meter.Float64Histogram(
+		pkgName+"/latency",
+		metric.WithDescription("Latency distribution of method calls"),
+	)
+	if err != nil {
+		otel.Handle(err)
+	}
+
+	completedCallsCounter, err = meter.Int64Counter(
+		pkgName+"/completed_calls",
+		metric.WithDescription("Count of completed method calls"),
+	)
+	if err != nil {
+		otel.Handle(err)
+	}
+}
+
+// providerName returns the name of the provider associated with the driver value.
+// It is intended to be used for metrics and tracing.
+func providerName(driver any) string {
+	// Return the package path of the driver's type.
+	if driver == nil {
+		return ""
+	}
+	t := reflect.TypeOf(driver)
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+	return t.PkgPath()
+}
+
 // Encrypt encrypts the plaintext and returns the cipher message.
 func (k *Keeper) Encrypt(ctx context.Context, plaintext []byte) (ciphertext []byte, err error) {
-	ctx = k.tracer.Start(ctx, "Encrypt")
-	defer func() { k.tracer.End(ctx, err) }()
+	start := time.Now()
+	ctx, span := k.tracer.Start(ctx, "Encrypt")
+	// Set span attributes for testing
+	span.SetAttributes(
+		attribute.String("gocdk.package", pkgName),
+		attribute.String("gocdk.method", "Encrypt"),
+	)
+	defer func() {
+		// Set status on span before ending
+		if err != nil {
+			span.SetAttributes(attribute.String("gocdk.status", "13")) // Internal error
```

Comment:
Where did you get 13? Isn't there a constant for it somewhere?

Reply:
This is cleaned up